src/controls/PolyControl.ts

```diff
@@ -0,0 +1,153 @@
+   * and the current position in canvas coordinate `transform.target` is a reference to the
+   * current object being transformed.
+   */
+  static polyActionHandler(
+    eventData: TPointerEvent,
+    transform: Transform,
+    x: number,
+    y: number
+  ) {
+    const poly = transform.target as Polyline,
+      currentControl = poly.controls[poly.__corner] as PolyControl,
+      mouseLocalPosition = getLocalPoint(transform, 'center', 'center', x, y),
+      polygonBaseSize = PolyControl.getSize(poly),
+      size = poly._getTransformedDimensions(),
+      sizeFactor = polygonBaseSize.divide(size),
+      shear = new Point(
+        Math.tan(degreesToRadians(poly.skewX)),
+        Math.tan(degreesToRadians(poly.skewY))
+      ),
+      adjustFlip = new Point(poly.flipX ? -1 : 1, poly.flipY ? -1 : 1);
+
+    const skewedPathOffset = PolyControl.applySkew(poly.pathOffset, shear),
+      finalPointPosition = PolyControl.removeSkew(
+        mouseLocalPosition
+          .multiply(adjustFlip)
+          .multiply(sizeFactor)
+          .add(skewedPathOffset),
+        shear
+      );
+
+    poly.points[currentControl.pointIndex] = finalPointPosition;
+    poly.setDimensions();
+
+    return true;
+  }
+
+  /**
+   * Keep the polygon in the same position when we change its `width`/`height`/`top`/`left`.
+   */
+  static anchorWrapper(anchorIndex: number, fn: TransformActionHandler) {
+    return function (
+      eventData: TPointerEvent,
+      transform: Transform,
+      x: number,
+      y: number
+    ) {
+      const poly = transform.target as Polyline,
+        absolutePoint = new Point(
+          poly.points[anchorIndex].x - poly.pathOffset.x,
+          poly.points[anchorIndex].y - poly.pathOffset.y
+        ).transform(poly.calcTransformMatrix()),
+        actionPerformed = fn(eventData, transform, x, y),
+        polygonBaseSize = PolyControl.getSize(poly),
+        shear = new Point(
+          Math.tan(degreesToRadians(poly.skewX)),
+          Math.tan(degreesToRadians(poly.skewY))
+        ),
+        adjustFlip = new Point(poly.flipX ? -1 : 1, poly.flipY ? -1 : 1);
+
+      const newPosition = PolyControl.applySkew(
+        new Point(
+          poly.points[anchorIndex].x,
+          poly.points[anchorIndex].y
+        ).subtract(poly.pathOffset),
+        shear
+      )
+        .divide(polygonBaseSize)
+        .multiply(adjustFlip);
+
+      poly.setPositionByOrigin(
+        absolutePoint,
+        newPosition.x + 0.5,
+        newPosition.y + 0.5
+      );
+      return actionPerformed;
+    };
+  }
+
+  static createPolyControls(numOfControls: number, options?: Partial<Control>) {
+    const lastControl = numOfControls - 1,
+      controls = {} as Record<string, PolyControl>;
+    for (let idx = 0; idx < numOfControls; idx++) {
+      controls[`p${idx}`] = new PolyControl(
+        {
+          actionHandler: PolyControl.anchorWrapper(
+            idx > 0 ? idx - 1 : lastControl,
```

Comment:
I gave this a bit of thought.
All the other vertexes remain in place so it doesn't really matter which one we anchor to, right?
If so we don't really need to pass this information and make `polyActionHandler` decide to which point to anchor by itself.
Furthermore, it doesn't depend on the number of vertexes...

Reply:
take a look d9195a406
https://codesandbox.io/s/custom-control-poly-fix-flip-forked-3k408w?file=/src/index.js

Reply:
well noted, nice!